monkestation/code/datums/components/regenerator.dm

```diff
@@ -0,0 +1,12 @@
+/datum/component/regenerator/RegisterWithParent()
+	. = ..()
+	RegisterSignals(parent, COMSIG_LIVING_ADJUST_ALL_DAMAGE_TYPES, PROC_REF(on_adjust_damage))
```

Comment:
we need to register **both** this _and_ `COMSIG_MOB_APPLY_DAMAGE` to ensure it catches all incoming damage.